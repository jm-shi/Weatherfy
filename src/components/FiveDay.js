const React = require('react');
const Temperature = require('./Temperature');
const api = require('../utils/api');
const formatter = require('../utils/formatter.js');
const formatDate = formatter.formatDate;
const formatShortDate = formatter.formatShortDate;
const formatTime = formatter.formatTime;
const formatShortTime = formatter.formatShortTime;
const toCelsius = formatter.toCelsius;

class FiveDay extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            forecastData: null,
            isLoading: true,
            useFahrenheit: true
        }
        this.useFahrenheit = this.useFahrenheit.bind(this);
    }

    componentDidMount() {
        this.updateData(this.props.data);
    }

    componentWillReceiveProps(nextProps) {
        this.updateData(nextProps.data);
    }

    useFahrenheit(result) {
        this.setState({ useFahrenheit: result })
    }

    updateData(zipcode) {
        api.fetchForecast(zipcode).then((data) =>
            this.setState(() => ({
                forecastData: data,
                isLoading: false
            })));
    }

    render() {
        const { isLoading, forecastData, useFahrenheit } = this.state;
        return (
            <div>
                {(isLoading || forecastData == null)
                ? <p style={{fontSize: '5vw'}}>Loading...</p>
                : <div className='weather-container'>
                    
                    <table>
                        <caption className='data-header' style={{marginBottom: '2vw'}}>{forecastData.city.name}</caption>
                        <tbody style={{textAlign: 'center'}}>
                            <tr className='five-day data-subheader'>
                                <th className='five-day data-details'></th>
                                <th className='five-day data-details'>Date</th>
                                <th className='five-day data-details'>Time</th>
                                <th className='five-day data-details'>Temperature</th>
                                <th className='five-day data-details conditions'>Conditions</th>
                            </tr>

                            {forecastData.list.map(item =>
                                <tr key={item.dt}>
                                    <td className='five-day data-details'><img src={`https://openweathermap.org/img/w/${item.weather[0].icon}.png`} /></td>
                                    <td className='five-day data-details short-date'>{formatShortDate(item.dt)}</td>
                                    <td className='five-day data-details long-date'>{formatDate(item.dt)}</td>
                                    <td className='five-day data-details short-time'>{formatShortTime(item.dt)}</td>
                                    <td className='five-day data-details long-time'>{formatTime(item.dt)}</td>
                                    <td className='five-day data-details'>{useFahrenheit 
                                        ? `${Math.round(item.main.temp)}°F`
                                        : `${Math.round(toCelsius(item.main.temp))}°C`}</td>
                                    <td className='five-day data-details conditions'>{item.weather[0].description}</td>
                                </tr>)
                            }
                        </tbody>
                    </table>
                </div>
                }
                <Temperature useFahrenheit={this.useFahrenheit} />
            </div>
        )
    }
}

module.exports = FiveDay;