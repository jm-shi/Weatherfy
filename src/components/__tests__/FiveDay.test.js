import React from 'react';
import FiveDay from '../FiveDay.js';
import { shallow } from 'enzyme';

test('should render FiveDay correctly', () => {
  const wrapper = shallow(<FiveDay />);
  expect(wrapper).toMatchSnapshot();
});
