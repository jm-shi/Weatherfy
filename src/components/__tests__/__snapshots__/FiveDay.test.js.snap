// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FiveDay correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FiveDay />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          style={
            Object {
              "fontSize": "5vw",
            }
          }
        >
          Loading...
        </p>,
        <Temperature
          useFahrenheit={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
          "style": Object {
            "fontSize": "5vw",
          },
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "useFahrenheit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            style={
              Object {
                "fontSize": "5vw",
              }
            }
          >
            Loading...
          </p>,
          <Temperature
            useFahrenheit={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "style": Object {
              "fontSize": "5vw",
            },
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "useFahrenheit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
